### DISCUSSION TITLE
svg board fill attacks color in square board

### DISCUSSION THREAD

**Discussion by sugizo** [niklasf/python-chess#988] [Ideas]
**this code** 
```
board_svg = chess.svg.board(board = board, 
                            lastmove = move, 
                            check = king_attacked, 
                            arrows = [chess.svg.Arrow(from_square, 
                                                      to_square) ], 
                            fill = dict.fromkeys(board.attacks(to_square), 
                                                 'red'), 
                            style = """
                            .arrow {opacity: .8; }
                            """, 
                            colors = chosen_color)
```
**produce**
![image](https://user-images.githubusercontent.com/2418295/232631191-98ec86aa-0f84-48c5-98e0-89870dab7481.png)

as you can see, the square in chess board is all filled with red, to indicate the chess piece range target attack

**expectation**
the sqare is not fully covered but the red color covered half, like the check indicator (with opacity or whatever use in check colored)
**e.g.**
![image](https://user-images.githubusercontent.com/2418295/232631413-2e6ae0de-236b-4f52-94ed-70d80170ec22.png)

as you can see the example, so that the rook, the pawn covered with another color (blue, green or whatever, to indicate that it can can be attack by the queen chess piece)

**Comment by niklasf** [comment]
`red` specifies a solid color. Opacity is supported using syntax like `#cc0000cc` (2 hex characters each for red, green, blue, opacity).

**Reply by sugizo** [reply]
another way around, to have opacity can be achieve with rgba
`'rgba(255, 0, 255, .8)'`

but my main purpose at first is still want to using radial gradient like on check mate value with different color, perhaps you can provide every properties on chess.svg.board() a class name in the future, so that we can modify it by modify style properties
e.g.
```
chess.svg.board(board = board, 
                            lastmove = move, 
                            check = king_attacked, 
                            arrows = [chess.svg.Arrow(from_square, 
                                                      to_square) ], 
                            fill = dict.fromkeys(board.attacks(to_square) ), 
                            style = """
                            .arrow {opacity: .8; }
                            .check {opacity: .8; } # not sure about this because you call the id gradient on def in svg syntax (just food for thought)
                            .fill {opacity: .8; fill: 'red'; }
                            """, 
                            colors = chosen_color)

```

thanks and best regards
